schemars/src/json_schema_impls/primitives.rs

```diff
@@ -54,7 +54,33 @@ simple_impl!(PathBuf => String);
 
 simple_impl!(Ipv4Addr => String, "ipv4");
 simple_impl!(Ipv6Addr => String, "ipv6");
-simple_impl!(IpAddr => String, "ip");
+
+impl JsonSchema for IpAddr {
+    no_ref_schema!();
+
+    fn schema_name() -> String {
+        "IpAddr".to_string()
+    }
+
+    fn schema_id() -> Cow<'static, str> {
+        Cow::Borrowed("IpAddr")
+    }
+    
+    fn json_schema(gen: &mut SchemaGenerator) -> Schema {
+        let schema_object = SchemaObject {
+            instance_type: Some(SingleOrVec::Single(Box::new(InstanceType::String))),
+            subschemas: Some(Box::new(SubschemaValidation {
+                any_of: Some(vec![
+                    gen.subschema_for::<Ipv4Addr>(),
+                    gen.subschema_for::<Ipv6Addr>(),
+                ]),
+                ..Default::default()
+            })),
+            ..Default::default()
+        };
```

Comment:
You might consider this to label the subschemas; this makes it easier for code generators to infer what's going on.

```suggestion
        let schema_object = SchemaObject {
            instance_type: Some(SingleOrVec::Single(Box::new(InstanceType::String))),
            subschemas: Some(Box::new(SubschemaValidation {
                any_of: Some(vec![
                    SchemaObject {
                        metadata: Some(Box::new(Metadata {
                            title: Some("V4".into()),
                            ..Default::default()
                        })),
                        subschemas: Some(Box::new(SubschemaValidation {
                            all_of: Some(vec![gen.subschema_for::<Ipv4Addr>()]),
                            ..Default::default()
                        })),

                        ..Default::default()
                    }
                    .into(),
                    SchemaObject {
                        metadata: Some(Box::new(Metadata {
                            title: Some("V6".into()),
                            ..Default::default()
                        })),
                        subschemas: Some(Box::new(SubschemaValidation {
                            all_of: Some(vec![gen.subschema_for::<Ipv6Addr>()]),
                            ..Default::default()
                        })),

                        ..Default::default()
                    }
                    .into(),
                ]),
                ..Default::default()
            })),
            ..Default::default()
        };
```